libnd4j/include/graph/RandomGenerator.h

```diff
@@ -194,8 +194,22 @@ namespace nd4j {
 
         template <typename T>
         _CUDA_HD FORCEINLINE T RandomGenerator::relativeT(Nd4jLong index, T from, T to) {            
-            auto t = this->relativeT<T>(index);            
-            auto z = from + (t * (to - from));
+            auto t = this->relativeT<T>(index);
+            auto z = from + T(t * (to - from));
+            return z;
+        }
+
+        template <>
+        _CUDA_HD FORCEINLINE Nd4jLong RandomGenerator::relativeT(Nd4jLong index, Nd4jLong from, Nd4jLong to) {
+            auto t = this->relativeT<double>(index);
+            auto z = from + Nd4jLong(t * (to - from));
+            return z;
+        }
+
+        template <>
+        _CUDA_HD FORCEINLINE int RandomGenerator::relativeT(Nd4jLong index, int from, int to) {
+            auto t = this->relativeT<float>(index);
```

Comment:
Why not something like this:
```
auto t = this->relativeT<int>(index);
```
?

Reply:
float value from 0 to 1 multiply to range and convert to int. As result - int value distributed between from and to uniformly.